include/async/queue.hpp

```diff
@@ -11,8 +11,8 @@ namespace async {
 
 template<typename T>
 struct queue {
-	void put(T item) {
-		_queue.push(item);
+	void put(T &&item) {
+		_queue.push(std::forward<T>(item));
```

Comment:
This does nothing here since `item` is not a forwarding reference.
```suggestion
	template<typename... Args>
	void emplace(Args ...&&item) {
		_queue.push(std::forward<Args>(args)...);
```